ci/appveyor-cpp-build.bat

```diff
@@ -132,10 +132,17 @@ set ARROW_HOME=%CONDA_PREFIX%\Library
 @rem ARROW-3075; pkgconfig is broken for Parquet for now
 set PARQUET_HOME=%CONDA_PREFIX%\Library
 
+@rem Move tzdata to a non-standard location to test the
+@rem configurability of the timezone database path
+mkdir %USERPROFILE%\Downloads\test\tzdata
+move %USERPROFILE%\Downloads\tzdata %USERPROFILE%\Downloads\test\tzdata
+
 python setup.py develop -q || exit /B
 
 set PYTHONDEVMODE=1
 
+@rem Configure the path of the timesone database to a new location
+python -c "import pyarrow;import os;path = os.path.expandvars(r'%USERPROFILE%\Downloads\test\tzdata');pa.set_timezone_db_path(path)"
```

Comment:
```suggestion
python -c "import pyarrow;import os;path = os.path.expandvars(r'%USERPROFILE%\Downloads\test\tzdata');pyarrow.set_timezone_db_path(path)"
```

Reply:
I also think this will not persist to the actual pytest invocation (that's a new python session, so (py)arrow will be a fresh import)

So we probably have to add a way to configure this with an env variable as well? Or a pytest config variable to pass to the `pytest` cmd? (do you know how R tests this?)

Reply:
In R it seems the wrapper around `timezone_db_path` is used to configure the path to show to the installation of the R `tzdb` package:

- https://tzdb.r-lib.org/reference/tzdb_initialize.html
-  https://tzdb.r-lib.org/reference/tzdb_path.html

and this package it is required for Windows systems.

In comparison, we want the wrapper function to be used / path to be configurable by the user. So yes, I guess a better option would be to set the path with env var. And then configure the location of the database in the `pyarrow/__init__`?